### DISCUSSION TITLE
Unet class activation map

### DISCUSSION THREAD

**Discussion by JumpLK** [Project-MONAI/MONAI#1657] [Q&A] [answered]
I'm using a Unet for 3D segmentation and I would like to know where it pays attention.
I would like to use the CAM or the GradCAM but I don't know what layer I can pass or should pass as arguments.

I tried blindly to fill those with for exemple :
cam = monai.visualize.GradCAM(nn_module=model, target_layers="layer4")
cam = monai.visualize.GradCAM(nn_module=model, target_layers="class_layers.prelu")
but it's not working so I would like to know how to find the target_layers (or function layers) to pass as arguments.

**Comment by rijobro** [comment, marked as answer]
This is my normal tactic:

```python
net = UNet(
    dimensions=2,
    in_channels=1,
    out_channels=3,
    channels=(4, 8),
    strides=(2,),
)
for name, _ in net.named_modules():
    print(name)
```

Which gives the output:

```
model
model.0
model.0.conv
model.0.adn
model.0.adn.N
model.0.adn.D
model.0.adn.A
model.1
model.1.submodule
model.1.submodule.conv
model.1.submodule.adn
model.1.submodule.adn.N
model.1.submodule.adn.D
model.1.submodule.adn.A
model.2
model.2.conv
```

So perhaps `model.2.conv`?

**Reply by rijobro** [reply]
To compare this the densenet output:

```
features
features.conv0
features.norm0
features.relu0
features.pool0
features.denseblock1
features.denseblock1.denselayer1
features.denseblock1.denselayer1.layers
...
features.denseblock4.denselayer16.layers.conv1
features.denseblock4.denselayer16.layers.norm2
features.denseblock4.denselayer16.layers.relu2
features.denseblock4.denselayer16.layers.conv2
features.norm5
class_layers
class_layers.relu
class_layers.pool
class_layers.flatten
class_layers.out
```

**Comment by JumpLK** [comment]
Yes thank you it works !

**Reply by rijobro** [reply]
But you're right that it's not immediately obvious. Perhaps we could enhance it so that you don't need to know layer names? @wyli

**Reply by wyli** [reply]
perhaps could enhance the docstring with this method `for name, _ in model.named_modules(): print(name)`?
